server/src/main/java/org/elasticsearch/common/unit/RelativeByteSizeValue.java

```diff
@@ -103,4 +108,27 @@ public String getStringRep() {
             return absolute.getStringRep();
         }
     }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeBoolean(isAbsolute());
+        if (isAbsolute()) {
+            assert absolute != null;
+            absolute.writeTo(out);
+        } else {
+            assert ratio != null;
+            out.writeDouble(ratio.getAsPercent());
+        }
+    }
+
+    public static RelativeByteSizeValue readFrom(StreamInput in) throws IOException {
+        RelativeByteSizeValue relativeByteSizeValue;
+        boolean isAbsolute = in.readBoolean();
+        if (isAbsolute) {
+            relativeByteSizeValue = new RelativeByteSizeValue(ByteSizeValue.readFrom(in));
+        } else {
+            relativeByteSizeValue = new RelativeByteSizeValue(new RatioValue(in.readDouble()));
+        }
+        return relativeByteSizeValue;
```

Comment:
nit: I'd suggest inlining these local variables:

```suggestion
        if (in.readBoolean()) {
            return new RelativeByteSizeValue(ByteSizeValue.readFrom(in));
        } else {
            return new RelativeByteSizeValue(RatioValue.readFrom(in));
        }
```